### DISCUSSION TITLE
postUpgradeTasks is not using node version specified in package.json

### DISCUSSION THREAD

**Discussion by vietanhtran16** [renovatebot/renovate#16350] [Suggest an Idea]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.64.9

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

We are trying to set up `postUpgradeTasks` in `renovate.json5` but been running into this issue
```
Command failed: yarn renovate-post-upgrade
error unified-home-monorepo@0.0.0-private: The engine "node" is incompatible with this module. Expected version "^14.18.2". Got "16.15.0"
error Commands cannot run with an incompatible environment.
```

In our `package.json`, we are specifying the node version like below and based on [the doc](https://docs.renovatebot.com/node/#file-support), renovate should respect and use that.
```
"engines": {
    "node": "^14.18.2"
  },
```

We also tried to force the version via renovate config but no luck too
```
"force": {
    "constraints": {
      "node": "^14.18.2"
    }
  }
```

One thing special about our repo is that it is a monorepo with utilities packages as well as the actually application itself. Not sure if that makes a difference but thought I should mention it. I saw another repo in our company that got this `postUpgradeTasks` working but they are not specifying the `engines` field so it wont complain also not a monorepo like us

If there is any other information you would like, I am more than happy to share

### Relevant debug logs

<details><summary>Logs</summary>

```
Command failed: yarn renovate-post-upgrade
error unified-home-monorepo@0.0.0-private: The engine "node" is incompatible with this module. Expected version "^14.18.2". Got "16.15.0"
error Commands cannot run with an incompatible environment.

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
Not a bug, currently by design
- #8111

**Reply by vietanhtran16** [reply]
From the link you shared, is that correct to understand that when renovate upgrades dependencies, it uses and honor the correct node versions in `package.json` but when `postUpgradeTasks` is ran, it uses the node version of the docker image instead?

**Reply by rarkins** [reply]
The answer partly depends on exactly how you are running Renovate

**Reply by vietanhtran16** [reply]
it is self-hosted at the moment if that is what you asking?

**Reply by vietanhtran16** [reply]
oh sorry actually we are using renovate github action instead `https://github.com/renovatebot/github-action` sorry for the confusion? @rarkins. How would I be able to enforce node version for post upgrade tasks in this case? Thanks

**Reply by vietanhtran16** [reply]
Looks like it is also using docker behind the scen so I guess same problem as mentioned in the issue above

**Reply by viceice** [reply]
yes, post upgrade tasks are always using global installed tools. they can't respect any constraints. follow the mentioned issue for tracking future improvements

**Reply by vietanhtran16** [reply]
awesome thanks for your clarification 👍 I will keep an eye on it
